tests/test_server.py

```diff
@@ -24,23 +19,48 @@ def negotiate_auth():
 def negotiate_and_ntlm_auth():
     return get_auth_response('NTLM', advertise_nego_and_ntlm=True)
 
-def get_auth_response(auth_type, advertise_nego_and_ntlm=False):
-    response_headers = {'WWW-Authenticate':auth_type if not advertise_nego_and_ntlm else 'Negotiate, NTLM'}
+# With NTLMv2 we need to mock out some functions that usually return random values to ensure we get the same message
+@mock.patch('os.urandom', side_effect=mock_random)
+@mock.patch('ntlm_auth.messages.get_version', side_effect=mock_version)
+@mock.patch("ntlm_auth.messages.get_random_export_session_key", side_effect=mock_random_session_key)
+@mock.patch('ntlm_auth.compute_response.get_windows_timestamp', side_effect=mock_timestamp)
+def get_auth_response(auth_type, mock_random, mock_version, mock_session_key, mock_timestamp, advertise_nego_and_ntlm=False):
+    # Create the NTLM context on the server for validation with requests_ntlm
+    ntlm_context = ntlm.Ntlm()
+    negotiate_response = ntlm_context.create_negotiate_message(domain.upper()).decode('ascii')
+
+    challenge_response = 'TlRMTVNTUAACAAAAAwAMADgAAAAzgoriASNFZ4mrze8AAAAAAAAAACQAJABEAAAABgBwFwAAAA9TAGUA' \
```

Comment:
The only issue with changing dependencies is that to get NTLMv2 authentication, the current method exposed by python-ntlm3 does not have enough data being passed in to calculate the full message. Even if we were able to make the changes in that library it wouldn't be backwards compatible. I am happy to try again but with the massive changes made to support these new features and after 3 months of hearing nothing has made me a bit weary about doing that.